### PR TITLE
Add go-vet to structured-merge-diff PR tests

### PR DESCRIPTION
**Pull request by apelisse**

### CODE REVIEW

File: config/jobs/kubernetes-sigs/structured-merge-diff/structured-merge-diff-presubmits.yaml [hunk 1]
```diff
@@ -19,6 +19,24 @@ presubmits:
...
+  - name: pull-structured-merge-diff-vet
+    labels:
+      preset-service-account: "true"
+    always_run: true
+    decorate: true
+    path_alias: sigs.k8s.io/structured-merge-diff
+    spec:
+      containers:
+      - image: gcr.io/k8s-staging-test-infra/kubekins-e2e:v20220729-07f065e00e-master
+        command:
```
Comment (BenTheElder) [inline config/jobs/kubernetes-sigs/structured-merge-diff/structured-merge-diff-presubmits.yaml:31]: FWIW: I recommend using a make target or script in the repo so you can:
- reproduce locally with lower effort
- update the command in the repo (and fix new failures in that PR) versus PRing test-infra

**Comment by BenTheElder** [comment]
/lgtm
/approve
/hold
one suggestion

File: config/jobs/kubernetes-sigs/structured-merge-diff/structured-merge-diff-presubmits.yaml (hunk 1 above)
Comment (apelisse) [reply config/jobs/kubernetes-sigs/structured-merge-diff/structured-merge-diff-presubmits.yaml:31]: Yeah, that repo has a no-makefile, no-shell script policy ;-) It's entirely handled by go and I'd be super sad if I had to make a Makefile to run `go test` or `go vet`. I don't really mind the occasional PRs to that repo, to update the CI, I think that's fine. I do tend to agree that the CI configuration belong in the repo itself (à la Github workflow). I can change to that if that's easier.

Comment (BenTheElder) [reply config/jobs/kubernetes-sigs/structured-merge-diff/structured-merge-diff-presubmits.yaml:31]: IMHO no makefile no shell script is an excessive policy that prevents useful things like running tests trivially reproducibly between local and CI once you start to add useful options like emitting coverage data that affect the test results. Or for example pinning the go version. https://github.com/kubernetes-sigs/kind/blob/36c62825cdb75365d18d62370320554494db486b/Makefile#L34

But not my project 🤷‍♂️

Comment (BenTheElder) [reply config/jobs/kubernetes-sigs/structured-merge-diff/structured-merge-diff-presubmits.yaml:31]: You don’t *have* to run the makefile to go test in a project that has one, but you can, to conveniently match what will run in CI. It’s possible to fully support “just use go” and still checkin your CI targets etc.

**Pull request merged**